File: project 2 - loop/python/average.py

print("calculate you degree")
print("if u want to exit just write exit")

total_course = 0
number_course = 0

while True:
     course = input(f"enter you degree numer{number_course + 1}")
     if course == "exit":
          break
     
     try:
          course = float(course)
          number_course += 1
          total_course += course
     except ValueError:
          print("print number or exit")  
          continue   
if total_course > 0 :
    print(f"your average : {total_course / number_course}")
else: 
     print("invalid input")

File: project 2 - loop/php/average.php
<?php

$total_course = 0;
$number_course = 0;


if (isset($_GET['course'])) {
  
    $courses = explode(',', $_GET['course']);

    foreach ($courses as $course) {
        $course = trim($course); 

        
        if (strtolower($course) === "exit") {
            break;
        }

        if (is_numeric($course)) {
            $course = floatval($course);
            $number_course += 1;
            $total_course += $course;
        } else {
            echo "Please enter a valid number or type 'exit'.\n";
            continue;
        }
    }
}

if ($number_course > 0) {
    echo "Your average: " . ($total_course / $number_course) . "\n";
} else {
    echo "Invalid input\n";
}
?>